api/src/services/application-csv-export.service.ts

```diff
@@ -131,131 +141,120 @@ export class ApplicationCsvExporterService
               .join(',') + '\n',
           );
 
-          for (let i = 0; i < applications.length / 1000 + 1; i++) {
-            // grab applications 1k at a time
-            const paginatedApplications =
-              await this.prisma.applications.findMany({
-                include: {
-                  ...view.csv,
-                  demographics: queryParams.includeDemographics
-                    ? {
-                        select: {
-                          id: true,
-                          createdAt: true,
-                          updatedAt: true,
-                          ethnicity: true,
-                          gender: true,
-                          sexualOrientation: true,
-                          howDidYouHear: true,
-                          race: true,
-                        },
+          const promiseArray: Promise<string>[] = [];
+          for (let i = 0; i < applications.length; i += NUMBER_TO_PAGINATE_BY) {
+            promiseArray.push(
+              new Promise(async (resolve) => {
+                // grab applications NUMBER_TO_PAGINATE_BY at a time
+                const paginatedApplications =
+                  await this.prisma.applications.findMany({
+                    include: {
+                      ...view.csv,
+                      demographics: queryParams.includeDemographics
+                        ? {
+                            select: {
+                              id: true,
+                              createdAt: true,
+                              updatedAt: true,
+                              ethnicity: true,
+                              gender: true,
+                              sexualOrientation: true,
+                              howDidYouHear: true,
+                              race: true,
+                            },
+                          }
+                        : false,
+                    },
+                    where: {
+                      listingId: queryParams.listingId,
+                      deletedAt: null,
+                    },
+                    skip: i,
+                    take: NUMBER_TO_PAGINATE_BY,
+                  });
+
+                let row = '';
+                paginatedApplications.forEach((app) => {
+                  let preferences: ApplicationMultiselectQuestion[];
+                  csvHeaders.forEach((header, index) => {
+                    let multiselectQuestionValue = false;
+                    let parsePreference = false;
+                    let value = header.path.split('.').reduce((acc, curr) => {
+                      // return preference/program as value for the format function to accept
+                      if (multiselectQuestionValue) {
+                        return acc;
```

Comment:
How does line 184 ever evaluate to true?

Reply:
as we iterate through in the `reduce` first `parsePreference` can become true, and then if that becomes true `multiselectQuestionValue` becomes true

Reply:
as we iterate through with the `reduce()` first `parsePreference` (line 206) can become true, and then later `multiselectQuestionValue` (200) can become true